blockchain/blockdao/blob_store.go

```diff
@@ -189,10 +200,20 @@ func (bs *blobStore) putBlob(blob []byte, height uint64, txHash [][]byte) error
 	// write the blob data and height
 	b.Put(_blobDataNS, key, blob, "failed to put blob")
 	b.Put(_hashHeightNS, _writeHeight, key, "failed to put write height")
-	if err := bs.kvStore.WriteBatch(b); err != nil {
+}
+
+func (bs *blobStore) deleteBlob(k []byte, v []byte, b batch.KVStoreBatch) error {
```

Comment:
nit: maybe do `dx, err := deserializeBlobIndex(v)` outside and pass in `dx`, feels more natural